src/main.rs

```diff
@@ -115,7 +128,9 @@ fn main() {
         Config::default()
     };
     let flags = Flags::configure_from(&matches, &config).unwrap_or_else(|err| err.exit());
-    let core = Core::new(flags);
+    let mut core = Core::new(flags);
 
     core.run(inputs);
+
+    core.exit();
```

Comment:
`core.run` can return the `ExitStatus` and we can just directly call `std::process::exit` here at the end.